### PR TITLE
build: cmake: define SCYLLA_ENABLE_PREEMPTION_SOURCE for dev build

### PR DESCRIPTION
**Pull request by tchaikov**
in fabab2f4, we introduced preemption_source, and added `SCYLLA_ENABLE_PREEMPTION_SOURCE` preprocessor macro to enable opt-in the pluggable preemption check.

but CMake building system was not updated accordingly.

so, in this change, let's sync the CMake building system with `configure.py`.

---

it's a cmake change, hence no need to backport

### CODE REVIEW

**Comment by tchaikov** [comment]
this discrepancy was identified using https://gist.github.com/tchaikov/eb2cdb08c013506346cd0c476f5b45da .

**Comment by tchaikov** [comment]
once this change lands on master, i will switch the default building system to cmake.

**Review by avikivity — APPROVED** [review]

**Comment by tchaikov** [comment]
> #### 🔴 CI State: FAILURE
> 
> ✅ - [Build](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10645/) ✅ - [Container Test](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10645/) ✅ - [dtest](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release/10846/) ✅ - [dtest with topology changes](https://jenkins.scylladb.com/job/scylla-master/job/gating-dtest-release-with-consistent-topology-changes/2825/) ❌ - [Unit Tests](https://jenkins.scylladb.com/job/scylla-master/job/scylla-ci/10645/)
> #### Failed Tests (1/31530):
> 
>     * [background_reclaim](https://jenkins.scylladb.com//job/scylla-master/job/scylla-ci/10645/testReport/junit/boost/logalloc_test/background_reclaim) [🔍](https://github.com/scylladb/scylladb/issues?q=is:issue+is:open+background_reclaim)
> 
> 
> #### Build Details:
> 
>     * Duration: 5 hr 55 min
> 
>     * Builder: i-01afb94e4d80fc3e2 (m5ad.8xlarge)

#15677

**Comment by tchaikov** [comment]
@scylladb/scylla-maint hello maintainers, could you help queue this change?

**Pull request closed without merging**
